File: 2016 project/Assets/Scripts/Player/PLayer.js
#pragma strict

var MovementSpeed : float = 10;
var Speedlimit : float = 10;
var JumpHeight : float = 400;
var OnGround : boolean = false;

function Start () {

}

function Update () {
//movement via acceleration
	if(Input.GetKey("a")){ //left
		GetComponent.<Rigidbody2D>().AddForce(new Vector2(-MovementSpeed, 0));
	}
	if(Input.GetKey("d")){ //right
		GetComponent.<Rigidbody2D>().AddForce(new Vector2(MovementSpeed, 0));
	}
	//Speedlimit via velocity
	if(GetComponent.<Rigidbody2D>().velocity.x < -Speedlimit){
		GetComponent.<Rigidbody2D>().velocity.x = -Speedlimit;
	} else if(GetComponent.<Rigidbody2D>().velocity.x > Speedlimit){
		GetComponent.<Rigidbody2D>().velocity.x = Speedlimit;
	}
	//jump (via massive upwards acceleration)
	if(Input.GetKeyDown("space") && OnGround){
		GetComponent.<Rigidbody2D>().AddForce(new Vector2(0, JumpHeight));
	}
//end movement
//debugs
}

function OnCollisionEnter2D (other: Collision2D){
	if(other.gameObject.tag == "Ground"){
		OnGround = true; //if on ground, make true
	}
}

function OnCollisionExit2D (other: Collision2D){
	if(other.gameObject.tag == "Ground"){
		OnGround = false; //if leaving ground, make false
	}
}

function OnCollisionStay2D (other : Collision2D){
	if(other.gameObject.tag == "Ground"){
		OnGround = true; //Still on the ground.
	}
}

File: 2016 project/Assets/Scripts/Player/Player.cs
using UnityEngine;
using System.Collections;

public class Player : MonoBehaviour {

	float MovementSpeed = 10;
	float SpeedLimit = 10;
	float JumpHeight = 400;
	bool OnGround = false;

	void Start () 
	{
	
	}
	
	
	void Update ()
	{
	//movement via acceleration
		if(Input.GetKey("a")){
			GetComponent<Rigidbody2D>().AddForce(new Vector2(-MovementSpeed, 0));
		}
		if(Input.GetKey("d")){
			GetComponent<Rigidbody2D>().AddForce(new Vector2(MovementSpeed, 0));
		}
		//Speed limit via velocity
        var rb = GetComponent<Rigidbody2D>();

		if(GetComponent<Rigidbody2D>().velocity.x < -SpeedLimit)
		{
			rb.velocity = new Vector2(-SpeedLimit,rb.velocity.y);
		}
		if(GetComponent<Rigidbody2D>().velocity.x > SpeedLimit)
		{
            rb.velocity = new Vector2(SpeedLimit, rb.velocity.y);
		}
		//jump (via massive upwards acceleration)
		if(Input.GetKeyDown("space") && OnGround)
		{
			GetComponent<Rigidbody2D>().AddForce(new Vector2(0, JumpHeight));
		}
	//end movement
	}

	void OnCollisionEnter2D (Collision2D other)
	{
		if(other.gameObject.tag == "Ground")
		{
			OnGround = true; //Landing on ground
		}
	}

	void OnCollisionExit2D (Collision2D other)
	{
		if(other.gameObject.tag == "Ground")
		{
			OnGround = false; //Off ground
		}
	}
}
